modules/chat/src/utils/index.ts

```diff
@@ -23,6 +25,64 @@ export async function validateUsersInput(grpcSdk: ConduitGrpcSdk, users: any[])
       });
     }
   }
+  return usersToBeAdded;
+}
+
+export async function sendInvitations(users: any, sender: User, room: ChatRoom, url: string, config: any, grpcSdk: ConduitGrpcSdk) {
+
+  const roomId = room._id;
+  let retMessage: any [] = [];
+  for (const invitedUser of users) {
+    const invitationsCount = await InvitationToken.getInstance().countDocuments({
+      room: roomId,
+      receiver: invitedUser._id,
+      sender: sender._id,
+    });
+    if (room.participants.includes(invitedUser._id)) {
+      retMessage.push({
+        receiver: invitedUser._id,
+        message: 'Already member',
+      });
+      continue;
```

Comment:
What I meant to say is, there's no need to complete the loop nor move on to the next one.
Both this and the invitationsCount check below should immediately throw an error.

`throw new GrpcError(status.INVALID_ARGUMENT, 'users array contains existing member or invited user ids');`

We could technically find all the possible problems with the query and return the appropriate error message collection, but we don't do that anywhere else and it's mostly a waste of cycles as the request should fail anyway.